### PR TITLE
KTOR-2977 Add parameters to refreshTokens

### PR DESCRIPTION
**Pull request by rsinukov**

### CODE REVIEW

File: ktor-client/ktor-client-plugins/ktor-client-auth/common/src/io/ktor/client/plugins/auth/providers/BearerAuthProvider.kt [hunk 1]
```diff
@@ -24,17 +25,25 @@ public class BearerTokens(
     public val refreshToken: String
 )
 
+private typealias RefreshTokens =
+    suspend (client: HttpClient, response: HttpResponse, tokens: BearerTokens?) -> BearerTokens?
```
Comment (e5l) [inline ktor-client/ktor-client-plugins/ktor-client-auth/common/src/io/ktor/client/plugins/auth/providers/BearerAuthProvider.kt:29]: Could you tell me why do we have the client in parameters and not in the receiver?

Comment (e5l) [reply ktor-client/ktor-client-plugins/ktor-client-auth/common/src/io/ktor/client/plugins/auth/providers/BearerAuthProvider.kt:29]: 3 parameters is quite long, should we try to group it inside some context-like class?

**Review by e5l — CHANGES_REQUESTED** [review]

File: ktor-client/ktor-client-plugins/ktor-client-auth/common/src/io/ktor/client/plugins/auth/providers/BearerAuthProvider.kt (hunk 1 above)
Comment (rsinukov) [reply ktor-client/ktor-client-plugins/ktor-client-auth/common/src/io/ktor/client/plugins/auth/providers/BearerAuthProvider.kt:29]: 1. `client.get("update-token").body<Token>()` looks more ideomatic than just `get("update-token").body<Token>()`
2. Yes, let's create a class.

Comment (e5l) [reply ktor-client/ktor-client-plugins/ktor-client-auth/common/src/io/ktor/client/plugins/auth/providers/BearerAuthProvider.kt:29]: 1. agree

**Review by e5l — APPROVED** [review]

**Pull request merged**